mobile/native/src/max_quantity.rs

```diff
@@ -123,14 +189,50 @@ fn calculate_max_quantity(
         leverage,
     );
 
-    Decimal::try_from(max_quantity.floor()).expect("to fit into decimal")
+    Decimal::try_from((max_quantity + open_quantity).floor()).expect("to fit")
 }
 
 #[cfg(test)]
 mod tests {
     use super::*;
     use rust_decimal_macros::dec;
 
+    #[test]
+    fn test_calculate_max_quantity_with_open_quantity() {
+        let price = Decimal::new(22001, 0);
+
+        let max_coordinator_margin = Amount::from_sat(765_763);
+        let max_trader_margin = Amount::from_sat(747_499);
+
+        let trader_leverage = 2.0;
+        let coordinator_leverage = 2.0;
+        let order_matching_fee_rate = dec!(0.003);
+        let open_quantity = dec!(323);
+
+        let max_quantity = calculate_max_quantity(
+            price,
+            max_coordinator_margin,
+            max_trader_margin,
+            None,
+            coordinator_leverage,
+            trader_leverage,
+            order_matching_fee_rate,
+            Amount::from_sat(4459),
+            open_quantity,
+        );
+
+        assert_eq!(dec!(648), max_quantity);
```

Comment:
I would like to understand how you have come to this value. For me it's hard to derive this myself.
Can you add the formula, or explain it in prosa why this should be 648?